### PR TITLE
auto detect csv files

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[EmptyDatasetCannotBeImported für CSV](https://trello.com/c/3E1YVCmO/191-emptydatasetcannotbeimported-f%C3%BCr-csv)

File: services/src/handlers/datasets.rs [hunk 1]
```diff
@@ -555,15 +632,41 @@ fn detect_time_type(columns: &Columns) -> OgrSourceDatasetTimeType {
...
 
-fn detect_vector_type(layer: &Layer) -> Result<VectorDataType> {
-    let ogr_type = layer
-        .defn()
-        .geom_fields()
-        .next()
-        .context(error::EmptyDatasetCannotBeImported)?
-        .field_type();
+fn detect_vector_geometry(dataset: &Dataset) -> DetectedGeometry {
+    if let Ok(layer) = dataset.layer(0) {
```
Comment (ChristianBeilschmidt) [inline services/src/handlers/datasets.rs:636]: for each?

File: services/src/handlers/datasets.rs [hunk 2]
```diff
@@ -555,15 +632,41 @@ fn detect_time_type(columns: &Columns) -> OgrSourceDatasetTimeType {
...
+        if let Some(g) = layer.defn().geom_fields().next() {
```
Comment (ChristianBeilschmidt) [inline services/src/handlers/datasets.rs:637]: while-let?

File: services/src/handlers/datasets.rs [hunk 3]
```diff
@@ -555,15 +632,41 @@ fn detect_time_type(columns: &Columns) -> OgrSourceDatasetTimeType {
...
+                            s
+                        })
+                        .map(Into::into)
+                        .unwrap_or(SpatialReferenceOption::Unreferenced),
+                };
+            }
+        }
+    }
+
+    DetectedGeometry {
```
Comment (ChristianBeilschmidt) [inline services/src/handlers/datasets.rs:655]:
```suggestion
    // fallback type if no geometry was found
    DetectedGeometry {
```

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: services/src/handlers/datasets.rs (hunk 1 above)
Comment (michaelmattig) [reply services/src/handlers/datasets.rs]: 87b15682f1e70e0687e5cb55d2d11478bc28fcf7

File: services/src/handlers/datasets.rs (hunk 2 above)
Comment (michaelmattig) [reply services/src/handlers/datasets.rs]: 87b15682f1e70e0687e5cb55d2d11478bc28fcf7

File: services/src/handlers/datasets.rs (hunk 3 above)
Comment (michaelmattig) [reply services/src/handlers/datasets.rs:657]: 87b15682f1e70e0687e5cb55d2d11478bc28fcf7

**Pull request merged**